httpbp/middlewares.go

```diff
@@ -498,21 +499,31 @@ func getContentLength(w http.ResponseWriter) int64 {
 	return 0
 }
 
-// contentLengthRecorder is used to record the content length of whats written via the ResponseWriter.
-type contentLengthRecorder struct {
+// responeRecorder records the following:
+//   1. HTTP response code passed to a call to WriteHeader.
+//   2. the content length of whats written.
+type responseRecorder struct {
 	http.ResponseWriter
 
-	iobp.CountingSink
+	bytesWritten int
+	responseCode int
 }
 
-func (r *contentLengthRecorder) Write(b []byte) (int, error) {
-	_, err := r.CountingSink.Write(b)
-	if err != nil {
-		log.Warnf("err CountingSink.Write(b) %w\n", err)
-	}
-	return r.ResponseWriter.Write(b)
+func (rr *responseRecorder) Write(b []byte) (n int, err error) {
+	n, err = rr.ResponseWriter.Write(b)
+	rr.bytesWritten += n
+	return n, err
+}
+
+func (rr *responseRecorder) WriteHeader(code int) {
+	rr.ResponseWriter.WriteHeader(code)
+	rr.responseCode = code
+}
+
+func (r *responseRecorder) setContentLength() {
+	r.ResponseWriter.Header().Set(contentLengthHeader, strconv.Itoa(r.bytesWritten))
 }
 
-func (r *contentLengthRecorder) setContentLength() {
-	r.ResponseWriter.Header().Set(contentLengthHeader, strconv.Itoa(int(r.Size())))
+func (r *responseRecorder) getCode(err error) int {
```

Comment:
Don't leak "business logic" into `responseRecorder`.  It should be very dumb, just tracking what it receives.  Take the code it recorded and use it elsewhere instead.